src/DataHandling/Logs/DataDogAudit.php

```diff
@@ -165,22 +271,242 @@ protected function removeColumnIfOnlyUnchanged($diffArray)
+            ->join('c.source', 's')
+            ->where("c.action = 'insert'")
+            ->andWhere('s.fk = :cfId')->setParameter('cfId', $cfId)
+            ->andWhere("s.tbl = 'custom_fields'")
+        ;
+        $cfLogs = $cfQb->getQuery()->getResult();
+        $fieldId = null;
+        foreach ($cfLogs as $log) {
+            if (isset($log->getDiff()['fieldId'])) {
+                $fieldId = $log->getDiff()['fieldId']['new'];
+                break;
+            }
+        }
+
+        if (is_null($fieldId)) { // fallback
+            $fieldId = 'customfield_'.$cfId;
+        }
+        $this->instanceCache['customfield_labels'][$cfId] = $fieldId;
+
+        return $fieldId;
+    }
+
+    /**
+     * Filters the diff format of CustomFields.
+     *
+     * To be used in overwritten getAllVersionsDiff() after calling it from parent class.
+     * Rewrites the values as simple value instead of array.
+     *
+     * @param array $diff
+     *
+     * @return array modified $diff
+     */
+    protected function filterCustomFieldsAllVersionsDiff($diff)
+    {
+        if (empty($this->instanceCache['customfield_labels'])) {
+            return $diff;
+        }
+        $cfLabels = array_unique($this->instanceCache['customfield_labels']);
+        foreach ($diff as $key => $changeEntry) {
+            $changes = $changeEntry['changes'];
+            foreach ($cfLabels as $label) {
+                if (!isset($changes[$label])) {
+                    continue;
+                }
+                $value = isset($changes[$label][self::KEY_ADD]) ? $changes[$label][self::KEY_ADD] : '';
+                $diff[$key]['changes'][$label] = $value;
+            }
+        }
+
+        return $diff;
+    }
+
+    /**
+     * Returns doctrines metadata for the main entity.
+     *
+     * @return \Doctrine\ORM\Mapping\ClassMetadata
+     */
+    private function getEntitiesClassMetaData()
+    {
+        return $this->em->getClassMetadata($this->cache['class']);
+    }
+
+    /**
+     * Caches a value as current.
+     *
+     * @param Association $ass
+     * @param string      $value
+     */
+    private function setCachedAssociationValue(Association $ass, $value)
+    {
+        $this->instanceCache['assocCurrentValues'][$ass->getTbl()][$ass->getFk()] = $value;
+    }
+
+    /**
+     * Gets a cached current value, and optionally deletes it.
+     *
+     * @param Association $ass
+     * @param bool        $delete
+     *
+     * @return string the cached current value
+     */
+    private function getCachedAssociationValue(Association $ass, $delete)
+    {
+        if (!isset($this->instanceCache['assocCurrentValues'][$ass->getTbl()][$ass->getFk()])) {
+            return;
+        }
+        $value = $this->instanceCache['assocCurrentValues'][$ass->getTbl()][$ass->getFk()];
+        if ($delete) {
+            unset($this->instanceCache['assocCurrentValues'][$ass->getTbl()][$ass->getFk()]);
+        }
+
+        return $value;
+    }
```

Comment:
The whole data is now more then 500 lines, what makes it too big to make changes easily.